### PR TITLE
fix(kafka): ConsumeClaim should not return err when retry recovery failed

### PR DESCRIPTION
**Pull request by zcong1993**
ConsumeClaim should not return err when retry recovery failed.

# Description

`ConsumeClaim` return error will cause restart loop of consume, if initialOffset is oldest, this message will be consumed forever. So we should log error and let it go like no retry does.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: internal/component/kafka/consumer.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func (consumer *consumer) ConsumeClaim(session sarama.ConsumerGroupSession, clai
 			}, func() {
 				consumer.k.logger.Infof("Successfully processed Kafka message after it previously failed: %s/%d/%d [key=%s]", message.Topic, message.Partition, message.Offset, asBase64String(message.Key))
 			}); err != nil {
-				return err
+				consumer.k.logger.Errorf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v.", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
```
Comment (ItalyPaleAle) [inline internal/component/kafka/consumer.go:47]: To differentiate this error message, can you please log the number of failed attempts too? For example:

`Failed processing Kafka message after %d attempts: %s/%d/%d [key=%d]. Error: %v.`

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Thanks for catching this! I agree this is more correct. However, could you please modify the error message a bit?

File: internal/component/kafka/consumer.go (hunk 1 above)
Comment (zcong1993) [reply internal/component/kafka/consumer.go:47]: > To differentiate this error message, can you please log the number of failed attempts too? For example:
> 
> `Failed processing Kafka message after %d attempts: %s/%d/%d [key=%d]. Error: %v.`

The `retry.NotifyRecover` seems can not get attempts times when failed.

Comment (zcong1993) [reply internal/component/kafka/consumer.go:47]: How about this `Failed processing Kafka message after it previously failed: %s/%d/%d [key=%d]. Error: %v.` ?

Comment (ItalyPaleAle) [reply internal/component/kafka/consumer.go:47]: Sounds good, but if I can wordsmith a bit: "Too many failed attempts at processing Kafka message." (+ the rest)

Comment (zcong1993) [reply internal/component/kafka/consumer.go]: done

**Review by ItalyPaleAle — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Pull request merged**